run/helm/snap/templates/daemonset.yaml

```diff
@@ -0,0 +1,43 @@
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: snap
+  namespace: {{ .Release.Namespace }}
+  labels:
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+spec:
+  template:
+    metadata:
+      name: snap
+      labels:
+        daemon: snapteld
+    spec:
+      containers:
+      - name: {{ .Chart.Name }}
+        image: "{{ .Values.image.repo }}:{{ .Values.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        securityContext:
+          privileged: {{ .Values.privileged }}
+        ports:
+        - containerPort: 8181
+        volumeMounts:
+        - name: autoload
+          mountPath: {{ .Values.config.autodiscover_path }}
+        - name: config
+          mountPath: /etc/snap
+        {{- range $key, $val := .Values.mounts }}
+        - name: {{ $key | replace "/" "" | replace "." "" }}
+          mountPath: {{ $val | quote }}
+        {{- end }}
+      volumes:
+      - name: autoload
+        hostPath:
```

Comment:
I'll explain why I used hostPath here. What I want to achieve are two things:
* Loading of plugins should be as simple as specifying **ONLY** binaries urls in the values file.
* The user should have a possibility to mount one or more custom task files inside the container (in a way that allows to mount different tasks/plugins per node - sometimes the user may want to monitor different metrics on different nodes)

Main advantage of this solution is that it makes this Chart very elastic. It also solves problems that exist in other solutions (emptyDir and persistent volume claim). 
1. No redownloading (emptyDir)- The user may download all plugins they need for the first time, then remove URLs from values and use plugins that are already downloaded in hostPath.
2. Using more than one custom tasks (emptyDir) - With autoload feature you may download only tasks that have URLs, that means exemplary tasks from Snap plugin repos. In almost all cases user will want to use their own task manifest/s not the ones available in Snap repo examples. With hostPath the user may provide his own plugin binaries and tasks and mount them inside the container.
3. I may be wrong, but don't you need gce or Ceph, GlusterFS-like storage configured to use persistent volume claims? And it would be really difficult to support all cases?